librz/core/p/core_java.c

```diff
@@ -1935,15 +1624,67 @@ static bool rz_cmd_java_handle_print_exceptions(RzCore *core, const char *input)
 			free(class_info);
 		}
 	}
-	return true;
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_java_handler(RzCore *core, int argc, const char **argv) {
+	for (int i = 0; i < END_CMDS; i++) {
+		if (!strncmp(argv[1], JAVA_CMDS[i].name, JAVA_CMDS[i].name_len)) {
+			return JAVA_CMDS[i].handler(core, argc - 2, argc > 2 ? &argv[2] : NULL);
+		}
+	}
+	return RZ_CMD_STATUS_NONEXISTINGCMD;
+}
+
+static const char *cmd_java_subcmd_choices[] = { "help", "set_flags", "prototypes", "resolve_cp", "calc_flags", "flags_str_at", "flags_str", "m_info", "f_info", "find_cp_const", "find_cp_value", "replace_cp_value", "replace_classname_value", "reload_bin", "summary", "exc", "calc_sz", "is_valid", NULL };
+static const RzCmdDescArg cmd_java_args[3] = {
+	{
+		.name = "subcmd",
+		.type = RZ_CMD_ARG_TYPE_CHOICES,
+		.optional = false,
```

Comment:
I couldn't find an easy way to understand what subcommands were accepted. I suggest you either:
- make this argument optional and assume `help` as subcommands when nothing is specified. You could make this clear by also using the `default_value` field.
- provide a `description` in the `java` cmd descriptor to describe at least how to get additional help.